draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -0,0 +1,546 @@
+---
+title: OAuth 2.0 Client Assertion in Workload Environments
+abbrev: Workload Identity
+docname:  draft-ietf-wimse-client-assertion-in-workload-environments-latest
+category: info
+
+ipr: trust200902
+area: Security
+workgroup: WIMSE
+keyword: Internet-Draft
+
+stand_alone: yes
+pi:
+  rfcedstyle: yes
+  toc: yes
+  tocindent: yes
+  sortrefs: yes
+  symrefs: yes
+  strict: yes
+  comments: yes
+  inline: yes
+  text-list-symbols: -o*+
+  docmapping: yes
+
+author:
+ -
+      ins: B. Hofmann
+      name: Benedikt Hofmann
+      email: hofmann.benedikt@siemens.com
+      org: Siemens
+
+ -
+      ins: H. Tschofenig
+      name: Hannes Tschofenig
+      email: hannes.tschofenig@gmx.net
+      org: Siemens
+
+ -
+      ins: E. Giordano
+      name: Edoardo Giordano
+      email: edoardo.giordano@nokia.com
+      org: Nokia
+
+ -
+      ins: Y. Rosomakho
+      name: Yaroslav Rosomakho
+      email: yrosomakho@zscaler.com
+      org: Zscaler
+
+ -
+      ins: A. Schwenkschuster
+      name: Arndt Schwenkschuster
+      email: arndts.ietf@gmail.com
+      org: SPIRL
+
+normative:
+  RFC2119:
+  RFC7521:
+  RFC7523:
+  RFC6749:
+  RFC8174:
+  RFC8414:
+  RFC7519:
+  RFC7517:
+  RFC8693:
+informative:
+  OIDC:
+     author:
+        org: Sakimura, N., Bradley, J., Jones, M., de Medeiros, B., and C. Mortimore
+     title: OpenID Connect Core 1.0 incorporating errata set 1
+     target: https://openid.net/specs/openid-connect-core-1_0.html
+     date: 8 November 2014
+  KubernetesServiceAccount:
+     title: Kubernetes Service Account
+     target: https://kubernetes.io/docs/concepts/security/service-accounts/
+     date: 10 May 2024
+  TokenReviewV1:
+     title: Kubernetes Token Review API V1
+     target: https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-review-v1/
+     date: 28 August 2024
+  TokenRequestV1:
+     title: Kubernetes Token Request API V1
+     target: https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/
+     date: 28 August 2024
+
+--- abstract
+
+The use of the OAuth 2.0 framework for container orchestration systems poses a challenge as managing secrets, such as client_id and client_secret, can be complex and error-prone. Instead of manual provisioning these credentials the industry has moved to a federation-based approach where credentials of the underlying workload platform are used as assertions towards an OAuth authorization server leveraging the Client Assertion Flow {{RFC7521}}, in particular {{RFC7523}}.
+
+This specifications describes a meta flow in {{overview}}, gives security recommendations in {{recommendations}} and outlines concrete patterns in {{patterns}}.
```

Comment:
```suggestion
This specification describes a meta flow in {{overview}}, gives security recommendations in {{recommendations}} and outlines concrete patterns in {{patterns}}.
```

But also you don't normally dive down into sections in the abstract. This works better in the Intro.

Reply:
that guy makes a good point